sentinelsat/sentinel.py

```diff
@@ -909,6 +910,54 @@ def _download_online_retry(
 
         return ret_val
 
+    def download_quicklooks(self, products, directory_path="."):
+        """Download quicklooks for a list of products.
+
+        Takes a dict of product IDs: product data as input. This means that the return value of
+        query() can be passed directly to this method.
+
+        File names on the server are used for the downloaded images, e.g.
+        "S2A_MSIL1C_20200924T104031_N0209_R008_T35WMV_20200926T135405.jpeg".
+
+        Parameters
+        ----------
+        products : dict
+            Dict of product IDs, product data
+        directory_path : string
+            Parent directory for quicklooks directory, where the downloaded images will be stored
+
+        Returns
+        -------
+        dict[string, dict]
+            A dictionary containing the error of products where either
+            quicklook was not available or it had an unexpected content type
+        """
+
+        self.logger.info("Will download %d quicklooks", len(products))
+
+        outdir = Path(directory_path + "/quicklooks")
```

Comment:
Please use pathlib to construct the path instead of string concatenation. This ensures that it will work on all OS's.

Reply:
I changed this to use os.path since pathlib isnt in the standard library for older versions. In updated version I do avoid using string concatenation. Will commit together with other changes highlighted on here. Thanks for having a look!